### ISSUE TITLE
Unable to connect to Ray dashboard on remote machine.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#4201]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Locally using MacOS, Ray is running on an Ubuntu 16.04 instance on EC2.
- **Ray installed from (source or binary)**: pip installed the latest wheels from S3 
- **Ray version**: 0.7.0.dev0
- **Python version**: 3.6

On EC2, I installed Ray with `pip install ray/.whl/ray-0.7.0.dev0-cp36-cp36m-linux_x86_64.whl`.

Then I started Ray with `ray.init()` in IPython.

I saw 

```
======================================================================
View the dashboard at http://<private-ip-address>:8080/?token=e1c477a7ecb4ed05dd47f55be967fb895fe343a6713479f5
======================================================================
```

On my local machine, I attempted to forward the relevant port with

```
ssh -L 8080:localhost:8080 -i <key-file> ubuntu@<public-ip-address>
```

I do not see any warnings about the port already being in use.

 Then I attempt to go to `http://127.0.0.1:8080/?token=e1c477a7ecb4ed05dd47f55be967fb895fe343a6713479f5` and get `ERR_CONNECTION_RESET`.

In the terminal where I am forwarding the port, I see errors that say `channel 3: open failed: connect failed: Connection refused`.

I also tried opening port `8080` to all TCP traffic on the EC2 instance and then going to `http://<public-ip-address>:8080/?token=e1c477a7ecb4ed05dd47f55be967fb895fe343a6713479f5`, but that also fails to connect.

`ps aux | grep dashboard.py` confirms that the dashboard is running on the EC2 instance.

The dashboard logs show only

```
======== Running on http://<private-ip-address>:8080 ========
(Press CTRL+C to quit)
```

cc @ls-daniel any ideas what the issue might be or what to try?

**Comment by ls-daniel** [comment]
Port forwarding won't work I think because it doesn't bind to localhost. You'd have to do

`ssh -L 8080:private-ip-address:8080 -i key-file ubuntu@private-ip-address`

or similar

What happens if you try just forcing the ip to be 0.0.0.0? e.g. in `ray/python/ray/dashboard.py` set `self.ip = '0.0.0.0'` instead of the current socket.gethostbyname stuff?

You should then at least be able to connect via port forwarding.

I'm not sure about the EC2 port opening business; again there it sounds like maybe you have two assigned IP addresses to the box and it's only binding to one of them? Again binding to 0.0.0.0 would fix that

**Comment by ls-daniel** [comment]
This should be fixed in #4285 though I have not tested.

**Comment by robertnishihara** [comment]
Fixed by #4428.

**Comment by davideger** [comment]
With the above fix, the invocation you need to get access to the default ray dashboard via ssh tunneling works as follows:

```
ssh -L 8265:0.0.0.0:8265 user@remotemachine
$ python3
> import ray
> ray.init(include_dashboard=True)
2021-07-22 12:56:20,209 INFO services:1272 -- View the Ray dashboard at http://127.0.0.1:8265
> # do more stuff
```

Then in your local browser hit up `http://localhost:8265/`
